virtcontainers/sandbox.go

```diff
@@ -641,11 +641,22 @@ func (s *Sandbox) createCgroupManager() error {
 	if spec != nil {
 		cgroupPath = spec.Linux.CgroupsPath
 
-		// kata should rely on the cgroup created and configured by
-		// container engine *only* if actual container was
-		// marked *explicitly* as sandbox through annotations.
-		if !s.config.HasCRIContainerType {
-			resources = *spec.Linux.Resources
+		// Kata relies on the cgroup parent created and configured by the container
+		// engine, but sometimes the sandbox cgroup is not configured and the container
+		// may have access to all the resources, hence the runtime must constrain the
+		// sandbox and update the list of devices with the devices hotplugged in the
+		// hypervisor.
+		resources = *spec.Linux.Resources
```

Comment:
I know this PR has been already merged. But I would like to get some background about this change, in what circumstances are you seeing that the sandbox cgroup is not configured? 
So regardless if this is the sandbox or container type, we now apply constraints on resources?

Reply:
> in what circumstances are you seeing that the sandbox cgroup is not configured?

docker and podman

> So regardless if this is the sandbox or container type, we now apply constraints on resources?

only when `sandbox_cgroup_only=true`